azkaban-web-server/src/web/js/azkaban/view/project-modals.js

```diff
@@ -133,6 +135,51 @@ azkaban.ProjectDescriptionView = Backbone.View.extend({
     $.get(requestURL, requestData, successHandler, "json");
   },
 
+  handleUploadSettingEdit: function (evt) {
+    console.log("Edit Upload Setting");
+    var value = null;
+    if ($('#adhoc-upload').hasClass('editable-placeholder')) {
+      value = '';
+      $('#adhoc-upload').removeClass('editable-placeholder');
+    }
+    else {
+      value = $('#adhoc-upload').text();
+    }
+    $('#adhoc-upload-edit').attr("value", value);
+    $('#adhoc-upload').hide();
+    $('#adhoc-upload-form').show();
+  },
+
+  handleUploadSettingSave: function (evt) {
+    var newText = $('#adhoc-upload-edit').val();
+    if ($('#adhoc-upload-edit').hasClass('has-error')) {
+      $('#adhoc-upload-edit').removeClass('has-error');
+    }
+    var requestURL = contextURL + "/manager";
+    var requestData = {
+      "project": projectName,
+      "ajax": "changeUploadSetting",
+      "adhocUpload": newText
+    };
+    var successHandler = function (data) {
+      if (data.error) {
+        $('#adhoc-upload-edit').addClass('has-error');
+        alert(data.error);
+        return;
+      }
+      $('#adhoc-upload-form').hide();
+      if (newText !== '') {
+        $('#adhoc-upload').text(newText);
+      }
+      else {
+        $('#adhoc-upload').text('true or false');
+        $('#adhoc-upload').addClass('editable-placeholder');
+      }
+      $('#adhoc-upload').show();
+    };
+    $.get(requestURL, requestData, successHandler, "json");
```

Comment:
This operation is modifying the project so it should probably be a POST request.

Reply:
Yes technically it should be POST. but our backend ajax handling does not differentiate get or post. I've changed to post already.